callautomation-openai-sample-csharp/CallAutomationOpenAI/Program.cs

```diff
@@ -67,31 +99,96 @@
         {
-            Console.WriteLine($"Play failed event received for connection id: {playFailedEvent.CallConnectionId}. Hanging up call...");
-            await HandleHangupAsync(answerCallResult.CallConnection);
+            logger.LogInformation($"Play failed event received for connection id: {playFailedEvent.CallConnectionId}. Hanging up call...");
+            await answerCallResult.CallConnection.HangUpAsync(true);
+        });
+        client.GetEventProcessor().AttachOngoingEventProcessor<CallTransferAccepted>(answerCallResult.CallConnection.CallConnectionId, async (callTransferAcceptedEvent) =>
+        {
+            logger.LogInformation($"Call transfer accepted event received for connection id: {callTransferAcceptedEvent.CallConnectionId}.");
         });
+        client.GetEventProcessor().AttachOngoingEventProcessor<CallTransferFailed>(answerCallResult.CallConnection.CallConnectionId, async (callTransferFailedEvent) =>
+        {
+            logger.LogInformation($"Call transfer failed event received for connection id: {callTransferFailedEvent.CallConnectionId}.");
+            var resultInformation = callTransferFailedEvent.ResultInformation;
+            logger.LogError("Encountered error during call transfer, message={msg}, code={code}, subCode={subCode}", resultInformation?.Message, resultInformation?.Code, resultInformation?.SubCode);
+
+            await HandlePlayAsync(callTransferFailurePrompt,
+                       transferFailedContext, answerCallResult.CallConnection.GetCallMedia());
 
+        });
         client.GetEventProcessor().AttachOngoingEventProcessor<RecognizeCompleted>(answerCallResult.CallConnection.CallConnectionId, async (recognizeCompletedEvent) =>
         {
             Console.WriteLine($"Recognize completed event received for connection id: {recognizeCompletedEvent.CallConnectionId}");
             var speech_result = recognizeCompletedEvent.RecognizeResult as SpeechResult;
-
             if (!string.IsNullOrWhiteSpace(speech_result?.Speech))
             {
                 Console.WriteLine($"Recognized speech: {speech_result.Speech}");
-                var chatGPTResponse = await GetChatGPTResponse(speech_result?.Speech);
-                await HandleChatResponse(chatGPTResponse, answerCallResult.CallConnection.GetCallMedia(), callerId, logger);
+
+                if (await DetectEscalateToAgentIntent(speech_result.Speech, logger))
+                {
+                    await HandlePlayAsync(EndCallPhraseToConnectAgent,
+                               connectAgentContext, answerCallResult.CallConnection.GetCallMedia());
+                }
+                else
+                {
+                    var chatGPTResponse = await GetChatGPTResponse(speech_result.Speech);
+                    logger.LogInformation($"Chat GPT response: {chatGPTResponse}");
+                    Regex regex = new Regex(chatResponseExtractPattern);

```

Comment:
same here, can this be a json parse rather than a regex?

Reply:
As we discussed offline, with json prompt 40% of time its giving without json or json object with case sensitives. For now, will go with this, in future will tweak the prompt to get the json response.